src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Dapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Modules;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Localization;
+using Newtonsoft.Json;
+using Orchard.Queries.Sql.ViewModels;
+using Orchard.Tokens.Services;
+using YesSql;
+
+namespace Orchard.Queries.Sql.Controllers
+{
+    [Feature("Orchard.Queries.Sql")]
+    public class AdminController : Controller
+    {
+        private readonly IAuthorizationService _authorizationService;
+        private readonly IStore _store;
+        private readonly ITokenizer _tokenizer;
+        private readonly IStringLocalizer<AdminController> _stringLocalizer;
+
+        public AdminController(
+            IAuthorizationService authorizationService, 
+            IStore store, 
+            ITokenizer tokenizer,
+            IStringLocalizer<AdminController> stringLocalizer)
+        {
+            _authorizationService = authorizationService;
+            _store = store;
+            _tokenizer = tokenizer;
+            _stringLocalizer = stringLocalizer;
+        }
+
+        public Task<IActionResult> Query(string query, [FromServices] ITokenizer tokenizer)
+        {
+            query = String.IsNullOrWhiteSpace(query) ? "" : System.Text.Encoding.UTF8.GetString(Convert.FromBase64String(query));
+            return Query(new AdminQueryViewModel { DecodedQuery = query }, tokenizer);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Query(AdminQueryViewModel model, [FromServices] ITokenizer tokenizer)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageSqlQueries))
+            {
+                return Unauthorized();
+            }
+
+            if (String.IsNullOrWhiteSpace(model.DecodedQuery))
+            {
+                return View(model);
+            }
+
+            if (String.IsNullOrEmpty(model.Parameters))
+            {
+                model.Parameters = "{ }";
+            }
+            
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
```

Comment:
Nope, this way you can display intermediary ellapsed time, while keeping the watch running. Think lap times.